### PR TITLE
Move token sending to authorization header instead of body parameter

### LINKED ISSUE #1132: web-api token should be passed in header instead of body param
### Description

Slack's api is dropping support in Feb 2021 for `token` to be passed in as a body param. We should update web-api to always pass in token via header

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] enhancement (feature request)

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

#### Packages:

Select all that apply:

- [x] `@slack/web-api`

### PR DESCRIPTION
**Pull request by filmaj**
Fixes #1132

I also removed some trailing whitespace because that's the kind of nitpicky dude I am

### CODE REVIEW

File: packages/web-api/src/WebClient.spec.js [hunk 1]
```diff
@@ -521,9 +522,6 @@ describe('WebClient', function () {
             const file = parts.files[0];
             // the filename is picked up from the the ReadableStream since it originates from fs
             assert.include(file, { fieldname: 'someBinaryField', filename: 'train.jpg' });
-
-            assert.lengthOf(parts.fields, 1);
-            assert.deepInclude(parts.fields, { fieldname: 'token', value: token });
```
Comment (filmaj) [inline packages/web-api/src/WebClient.spec.js:526]: With the token no longer a part of the request body, this part needed removal.

File: packages/web-api/src/WebClient.ts [hunk 2]
```diff
@@ -183,6 +183,9 @@ export class WebClient extends Methods {
       this.logger = getLogger(WebClient.loggerName, logLevel ?? LogLevel.INFO, logger);
     }
 
+    // eslint-disable-next-line no-param-reassign
+    if (!headers.Authorization) headers.Authorization = `Bearer ${this.token}`;
```
Comment (seratch) [inline packages/web-api/src/WebClient.ts:187]: `this.token` can be absent. For instance, some API calls (such as oauth.v2.access with code parameter, openid.connect.token with code parameter, api.test) do not require token in the authorization header. Can you update the logic to check if `this.token` exists too?

File: packages/web-api/src/WebClient.spec.js [hunk 3]
```diff
@@ -425,7 +426,7 @@ describe('WebClient', function () {
     it('should properly serialize simple API arguments', function () {
       const scope = nock('https://slack.com')
         // NOTE: this could create false negatives if the serialization order changes (it shouldn't matter)
-        .post(/api/, 'token=xoxb-faketoken&team_id=T12345678&foo=stringval&bar=42&baz=false')
+        .post(/api/, 'team_id=T12345678&foo=stringval&bar=42&baz=false')
         .reply(200, { ok: true });
       return this.client.apiCall('method', { foo: 'stringval', bar: 42, baz: false, team_id: 'T12345678' })
```
Comment (seratch) [inline packages/web-api/src/WebClient.spec.js:431]: Can we override the token in the authorization header if a developer passes a token like `this.client.apiCall('method', { token: 'xoxb-' })`? Switching tokens is not a rare use case. For instance, the singleton `WebClient` does not hold any token and your app utilizes multiple user tokens for user related API calls.

**Review by seratch — CHANGES_REQUESTED** [review]
I like the change! Can you check the two points I mentioned in comments?

**Comment by filmaj** [comment]
@seratch I believe I have addressed your comments, and also added another test for the Authorization header overriding behaviour via `apiCall()` options parameter.

**Review by seratch — APPROVED** [review]
@filmaj LGTM! Having unit tests this way makes reviews easier 👍

**Review by stevengill — APPROVED** [review]
Looks great!

**Pull request merged**